### PR TITLE
Moving table generation into results_processing

### LINKED ISSUE #411: Move table generation out of results output and into its own file

### PR DESCRIPTION
**Pull request by wathen**
#### Description of Work

Fixes #411 

Moves table generation out of `core` and into `results_processing`

#### Testing Instructions

1. Check tables are consistent with master

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

File: fitbenchmarking/results_processing/tables.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+from collections import OrderedDict
+import copy
+import inspect
+from jinja2 import Environment, FileSystemLoader
+import os
+import pandas as pd
+
+import fitbenchmarking
+
+error_options = {0: "Successfully converged",
```
Comment (AndrewLister-STFC) [inline fitbenchmarking/results_processing/tables.py]: Constants are usually caps

Comment (wathen) [inline fitbenchmarking/results_processing/tables.py]:
```suggestion
ERROR_OPTIONS = {0: "Successfully converged",
```

File: fitbenchmarking/results_processing/tables.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        style_css = os.path.join(html_page_dir, 'style_sheet.css')
+        env = Environment(loader=FileSystemLoader(html_page_dir))
+        template = env.get_template("table_template.html")
+        output_file = file_path + 'html'
+
+        with open(output_file, "w") as f:
+            f.write(template.render(css_style_sheet=style_css,
+                                    result_name=title,
+                                    table=table_style.render(),
+                                    error_message=error_options))
```
Comment (wathen) [inline fitbenchmarking/results_processing/tables.py]:
```suggestion
                                    error_message=ERROR_OPTIONS))
```

File: fitbenchmarking/results_processing/tables.py (hunk 1 above)
Comment (wathen) [reply fitbenchmarking/results_processing/tables.py]: Corrected

File: fitbenchmarking/core/results_output.py [hunk 3]
```diff
@@ -4,35 +4,15 @@
...
-import copy
 import inspect
 import os
 
 from jinja2 import Environment, FileSystemLoader
-import pandas as pd
 
 import fitbenchmarking
-from fitbenchmarking.results_processing import plots, support_page
+from fitbenchmarking.results_processing import plots, tables, visual_pages
```
Comment (wathen) [inline fitbenchmarking/core/results_output.py]:
```suggestion
from fitbenchmarking.results_processing import plots, tables, support_page
```

**Pull request merged**